### PR TITLE
3163 refactor base mdm classes

### LINKED ISSUE #3163: implement mdm endpoints for admin-json - some code refactoring to help
**Describe the issue**
Some refactoring of code in MdmPControllerHelper and MdmProviderDstu3Plus can help make higher levels more extensible.

**Environment (please complete the following information):**
-version 5.7.0-PRE2-snapshot

### PR DESCRIPTION
**Pull request by TipzCM**
closes #3163

### CODE REVIEW

**Review by katiesmilecdr — APPROVED** [review]
Overall looks great! Only one question: moving MdmMatch helper methods from `MdmProviderDstu3Plus` to `MdmContollerHelper` is just for readability?

File: hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerHelper.java [hunk 1]
```diff
@@ -21,33 +21,55 @@
...
 	private final IResourceLoader myResourceLoader;
 	private final IMdmSettings myMdmSettings;
 	private final MessageHelper myMessageHelper;
+	private final IMdmMatchFinderSvc myMdmMatchFinderSvc;
 
 	@Autowired
-	public MdmControllerHelper(FhirContext theFhirContext, IResourceLoader theResourceLoader, IMdmSettings theMdmSettings, MessageHelper theMessageHelper) {
+	public MdmControllerHelper(FhirContext theFhirContext,
+										IResourceLoader theResourceLoader,
+										IMdmMatchFinderSvc theMdmMatchFinderSvc,
```
Comment (tadgh) [inline hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerHelper.java:66]: Will this change in the constructor break smile? does smile instantiate this bean anywhere? if so, you will need to bump the HAPI version. If not, all's well. Worth checking though.

**Review by tadgh — APPROVED** [review]
Approved pending a tiny check

File: hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerHelper.java (hunk 1 above)
Comment (TipzCM) [reply hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/provider/MdmControllerHelper.java:66]: No.
Only uses in my project (that combines both) are:
* MdmConsumerConfig (in fhir; this is the only place constructor is used)
* MdmControllerSvcImpl (as an imported bean - but also in fhir)

**Comment by TipzCM** [comment]
> Overall looks great! Only one question: moving MdmMatch helper methods from `MdmProviderDstu3Plus` to `MdmContollerHelper` is just for readability?

It was for readability and simplification. The move was needed for mdm-match (which is not needed anymore), but it still looks like cleaner code this way, so i left it

**Pull request merged**